src/Microsoft.AspNet.Mvc.Razor.Host/MvcRazorParser.cs

```diff
@@ -107,9 +125,43 @@ private class GlobalImportAddRemoveTagHelperVisitor : TagHelperDirectiveSpanVisi
                 IEnumerable<TagHelperDirectiveDescriptor> descriptors,
                 ParserErrorSink errorSink)
             {
-                return base.GetTagHelperDescriptorResolutionContext(
-                    _globalImportDirectiveDescriptors.Concat(descriptors),
-                    errorSink);
+                var directivesToImport = MergeDirectiveDescriptors(descriptors, _globalImportDirectiveDescriptors);
+
+                return base.GetTagHelperDescriptorResolutionContext(directivesToImport, errorSink);
+            }
+
+            private static IEnumerable<TagHelperDirectiveDescriptor> MergeDirectiveDescriptors(
+                IEnumerable<TagHelperDirectiveDescriptor> descriptors,
+                IEnumerable<TagHelperDirectiveDescriptor> inheritedDescriptors)
+            {
+                var mergedDescriptors = new List<TagHelperDirectiveDescriptor>();
+                TagHelperDirectiveDescriptor prefixDirectiveDescriptor = null;
+
+                foreach (var descriptor in inheritedDescriptors)
+                {
+                    if (descriptor.DirectiveType != TagHelperDirectiveType.TagHelperPrefix)
+                    {
+                        mergedDescriptors.Add(descriptor);
+                    }
+                    else
+                    {
+                        // Always take the latest @tagHelperPrefix descriptor. Can only have 1 per page.
+                        prefixDirectiveDescriptor = descriptor;
+                    }
+                }
+
+                // We need to see if the provided descriptors contain a @tagHelperPrefix directive. If so, it 
+                // takes presedence and overrides any provided by the inheritedDescriptors. If not we need to add the 
+                // inherited @tagHelperPrefix directive back into the merged list.
+                if (prefixDirectiveDescriptor != null &&
+                    !descriptors.Any(descriptor => descriptor.DirectiveType == TagHelperDirectiveType.TagHelperPrefix))
+                {
+                    mergedDescriptors.Add(prefixDirectiveDescriptor);
+                }
```

Comment:
:-1:  Not a special case, it enables the user to override the `@tagHelperPrefix` directive specified in the `_GlobalImport`. That's the expected inheritance behavior. We do similar things for `@model` where multiple on a single page is invalid but in a `_GlobalImport` hierarchy is valid because they override eachother.